### ISSUE TITLE
psfgen not available through conda 

### ISSUE THREAD

**Issue by H-EKE** [Acellera/htmd#1103]
Hi,

Im trying to use ```HTMD version (2.5.3+0.g155e4d4.dirty)``` to build a system and run it with ACEMD4.

I am trying to install also psfgen in the same conda environment but it is not working 

```
conda install psfgen -c acellera
```

```
Channels:
 - acellera
 - conda-forge
 - https://software.repos.intel.com/python/conda
 - salilab
 - defaults
Platform: linux-64
Collecting package metadata (repodata.json): done
Solving environment: done

added / updated specs:
    - psfgen


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    psfgen-2.0.0_3             |                0          81 KB  acellera
    ------------------------------------------------------------
                                           Total:          81 KB

The following NEW packages will be INSTALLED:

  psfgen             acellera/linux-64::psfgen-2.0.0_3-0 


Proceed ([y]/n)? y


Downloading and Extracting Packages:
psfgen-2.0.0_3       | 81 KB     |                                                                                                                 |   0% Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='conda.anaconda.org', port=443): Read timed out. (read timeout=60.0)")': /acellera/linux-64/psfgen-2.0.0_3-0.tar.bz2

                                                                                                                                                          

CondaHTTPError: HTTP 503 SERVICE UNAVAILABLE for url <https://conda.anaconda.org/acellera/linux-64/psfgen-2.0.0_3-0.tar.bz2>
Elapsed: 02:00.033016
CF-RAY: 8f4244ddabb17178-DUS

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.

CondaHTTPError: HTTP 503 SERVICE UNAVAILABLE for url <https://conda.anaconda.org/acellera/linux-64/psfgen-2.0.0_3-0.tar.bz2>
Elapsed: 02:00.033016
CF-RAY: 8f4244ddabb17178-DUS

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.

```

Any idea how I can install psfgen properly?

**Comment by stefdoerr** [comment]
This looks like an internet connection issue. Can you try again a bit later?

**Comment by H-EKE** [comment]
It is working now. Thanks!

It’s working now—thanks for the help! I have a couple of questions about ACEMD4:
	1.	License Handling
In ACEMD3, we could verify the commercial license by either exporting the IP or parsing the license.dat file using:

```
acemd3 --licence
# Licence:
#   Check floating licence:
#     ACELLERA_LICENCE_SERVER -- not defined
#     ACELLERA_LICENSE_SERVER -- GRANTED
#   Check node-locked licence:
#     ACELLERA_LICENCE_FILE -- not defined
#     ACELLERA_LICENSE_FILE -- not defined
#     /opt/acellera/licence.dat -- DENIED (Unable to locate target file)
#     /opt/acellera/license.dat -- DENIED (Unable to locate target file)
#     /home/.acellera/licence.dat -- DENIED (Unable to locate target file)
#     /home/.acellera/license.dat -- DENIED (Unable to locate target file)
#
```

But now with acemd4, I dont see how to do it, as each time that I use

```
acemd --license
```
I just got this
```
2024-12-19 14:20:18,809 - acemd - INFO - # You are on the latest ACEMD version (4.0.2).
```
Is there a way to explicitly check license details in ACEMD4 as we did before?

	2.	Handling Outputs and Restarting Simulations

Previously, in ACEMD3, I could redirect output logs and run simulations like this:
```

acemd --input input --ncpus \${SLURM_CPUS_PER_TASK}  >> production.log

```

In ACEMD4, the output is handled through a CSV file. Could you clarify:
	•	How output handling works in ACEMD4?
	•	What happens to the output files when a simulation is restarted? Are they overwritten or appended?

Thanks in advance for your help!

**Comment by stefdoerr** [comment]
**License**
this is weird, it works for me. Can you run `which acemd` to make sure you are using the right executable maybe? How did you install ACEMD 4?
```
(pyacemd)  sdoerr@yoga  ~/Work/parameterize  ↱ main  acemd --license
2024-12-19 15:37:02,617 - acemd - INFO - # You are on the latest ACEMD version (4.0.2).
# Licence:
#   Product: ACEMD 4.0
#   Check floating licence:
#     ACELLERA_LICENCE_SERVER -- not defined
#     ACELLERA_LICENSE_SERVER -- not defined
#   Check node-locked licence:
#     ACELLERA_LICENCE_FILE -- not defined
#     /opt/acellera/licence.dat -- DENIED (Unable to locate target file)
#     /opt/acellera/license.dat -- DENIED (Unable to locate target file)
#     /home/sdoerr/.acellera/licence.dat -- DENIED (Unable to locate target file)
#     /home/sdoerr/.acellera/license.dat -- DENIED (Unable to locate target file)
#     /home/sdoerr/slpm/license.dat -- GRANTED (expires in 321 days)
```

**Outputs / restarting**
You should still redirect the output as before. Indeed now it also outputs energy and progress related information in a CSV file for easy programmatic parsing and reading. But the CSV file doesn't contain information about the setup or the errors which is printed to the console so you should still keep that information in a file if running unsupervised.

When a simulation is restarted and you have `restart: true` in the config file it will read the `restart.chk` file loading all the state of the simulation from there and continue the simulation. It will append to all files, it will not overwrite them.

**Comment by H-EKE** [comment]
Thanks for your fast reply!

```
~/.conda/envs/acemd/bin/acemd
```

```
~/.conda/envs/acemd/bin/acemd --version
2024-12-19 14:45:52,898 - acemd - INFO - # You are on the latest ACEMD version (4.0.2).
ACEMD version 4.0.2
```

```
~/.conda/envs/acemd/bin/acemd --license
2024-12-19 14:46:11,465 - acemd - INFO - # You are on the latest ACEMD version (4.0.2)
```

I installed it like this

```
conda install acemd=4.0.2 cuda-version=11.8 -c acellera -c conda-forge -y
``

**Comment by stefdoerr** [comment]
I tried again with a totally fresh environment installing with your command and it prints the license message for me. Very weird. I assume this is Linux, not OSX, right? I haven't compiled the license checker for OSX yet.

**Comment by H-EKE** [comment]
I tied also a new environment, it still the same

```
conda install acemd cuda-version=12 -c acellera -c conda-forge
```

```
acemd --license
2024-12-19 14:54:57,388 - acemd - INFO - # You are on the latest ACEMD version (4.0.2).
(acemd4) [hnode02 ~]$ acemd --help   
2024-12-19 14:56:47,748 - acemd - INFO - # You are on the latest ACEMD version (4.0.2).
usage: acemd [-h] [--input INPUT] [--platform PLATFORM] [--device [DEVICE ...]] [--ngpus NGPUS] [--ncpus NCPUS] [--precision PRECISION] [--version]
             [--license]

options:
  -h, --help            show this help message and exit
  --input INPUT         The input file/directory
  --platform PLATFORM   The platform to use for the simulation
  --device [DEVICE ...]
                        The device indices to use for the simulation
  --ngpus NGPUS         The number of GPUs to use for the simulation
  --ncpus NCPUS         The number of CPU threads to use for the simulation
  --precision PRECISION
                        The precision to use for the simulation
  --version             Print the version and exit
  --license             Print the license and exit

```

Im using ```Rocky Linux release 8.7 (Green Obsidian)```

**Comment by H-EKE** [comment]
```
$CONDA_PREFIX/lib/python3.1/site-packages/acemd/share/license-checker  
/lib/python3.1/site-packages/acemd/share/license-checker: /lib64/libm.so.6: version `GLIBC_2.29' not found (required by /lib/python3.1/site-packages/acemd/share/license-checker)  
/lib/python3.1/site-packages/acemd/share/license-checker: /opt/ohpc/pub/compiler/gcc/9.4.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /lib/python3.1/site-packages/acemd/share/license-checker)  
/lib/python3.1/site-packages/acemd/share/license-checker: /lib64/libc.so.6: version `GLIBC_2.33' not found (required by /lib/python3.1/site-packages/acemd/share/license-checker)  
/lib/python3.1/site-packages/acemd/share/license-checker: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by /lib/python3.1/site-packages/acemd/share/license-checker)  

```

**Comment by stefdoerr** [comment]
Ok that explains it :D Thanks!

**Comment by stefdoerr** [comment]
[license-client.zip](https://github.com/user-attachments/files/18199547/license-client.zip)
Can you overwrite `$CONDA_PREFIX/lib/python3.1/site-packages/acemd/share/license-checker` with the file in this zip and try again please?

**Comment by stefdoerr** [comment]
Ah sorry rename it as well to license-checker from license-client

**Comment by stefdoerr** [comment]
Okay. I am out the next week on holidays but I'll take a look when I'm back. It doesn't affect you in any way though don't worry, it's meant to just notify you when the license expires, there are no limitations.

**Comment by H-EKE** [comment]
That works. Thanks a lot and wish you nice holidays!

**Comment by stefdoerr** [comment]
I figured it out. Thanks for the report! The license printing will be fixed in 4.0.3 next week.
